ios/RNTrackPlayer/RNTrackPlayerAudioPlayer.swift

```diff
@@ -79,7 +79,67 @@ public class RNTrackPlayerAudioPlayer: QueuedAudioPlayer {
 
 		self.reactEventEmitter.sendEvent(withName: "playback-state", body: ["state": state.rawValue])
     }
-    
+
+    override func AVWrapper(didReceiveMetadata metadata: [AVMetadataItem]) {
+        func getMetadataItem(forIdentifier:AVMetadataIdentifier) -> String? {
+            let data: String = AVMetadataItem.metadataItems(from: metadata, filteredByIdentifier: forIdentifier).first?.stringValue ?? ""
+            return data.isEmpty ? nil : data
+        }
+
+        super.AVWrapper(didReceiveMetadata: metadata)
+        var data : Dictionary<String, Optional<String>> = [
+            "source": nil,
+            "title": nil,
+            "url": nil,
+            "artist": nil,
+            "album": nil,
+            "date": nil,
+            "genre": nil,
+        ]
+        var source: String? {
+            switch metadata.first?.keySpace {
+            case AVMetadataKeySpace.id3:
+                return "id3"
+            case AVMetadataKeySpace.icy:
+                return "icy-headers"
+            case AVMetadataKeySpace.quickTimeMetadata:
+                return "quicktime"
+            case AVMetadataKeySpace.common:
+                return "common"
+            default: return "other"
```

Comment:
Is there really no way to know whether it is icy metadata?

Reply:
not that I have been able to find..

Reply:
icy metadata comes in like this: `[<AVMetadataItem: 0x2828a1650, identifier=common/title, keySpace=comn, key class = NSTaggedPointerString, key=title, commonKey=title, extendedLanguageTag=(null), dataType=(null), time={17280/44100 = 0.392}, duration={INVALID}, startDate=(null), extras={
}, value class=__NSCFString, value=Vanessa Wagner - Etude 9 (Nadia Struiwigh Experimental Edit)>]`

Reply:
That's unfortunate
What do you think about changing "common" and "other" to "unknown"?

Reply:
yes that makes sense - will do